redirect-rules/simple/rule.js

```diff
@@ -0,0 +1,36 @@
+function redirectToConsentForm (user, context, callback) {
+  var hasConsented = user.app_metadata && user.app_metadata.has_consented;
+
+  // redirect to consent form if user has not yet consented
+  if (!hasConsented && context.protocol !== 'redirect-callback') {
+    var auth0Domain = auth0.baseUrl.match(/([^:]*:\/\/)?([^\/]+\.[^\/]+)/)[2];
+
+    context.redirect = {
+      url: configuration.CONSENT_FORM_URL +
+        '&auth0_domain=' + encodeURIComponent(auth0Domain) +
+        // generate random state string that Auth0 will validate when we return
+        '&state=' + encodeURIComponent(new Buffer(Math.random().toString()).toString('base64'))
```

Comment:
I am not sure what the intent of the future implementation is, but I thought we were planning to add this state parameter automatically to the redirect URL at the pipeline level. In this context adding it at the app level may create a conflict. I am sure @jaredhanson has this figured out.